docs/source/documentation/concepts.rst

```diff
@@ -51,11 +51,18 @@ Gathering tasks into a single compute plan will lead to a more optimized compute
 
 Note that you can register a task alone, i.e. not put the task in a compute plan, but Substra will still create a compute plan for you for this specific task.
 
+.. _concept_task:
+
 Task
 ^^^^
 
 A task correspond to a computation task. The task can use any inputs (data, functions or output from other tasks). The function is expected to write the outputs in files, on paths given as outputs dictionary.
 
+Rank
+^^^^
+
+A rank is the order of a task within a compute plan. A task of rank ``n`` will depend on the output of tasks of rank ``n-1``.
+
```

Comment:
I have always struggle toto properly define the rank. "Order" is a bit misleading I think as it's not it is not necessarily the order in which tasks will be processed in the backend.

I think one way to define it is to use the graph / CP analogy. So [wiki says](https://en.wikipedia.org/wiki/Tree_(graph_theory)#Rooted_tree:~:text=The%20height%20of%20a%20vertex%20in%20a%20rooted%20tree%20is%20the%20length%20of%20the%20longest%20downward%20path%20to%20a%20leaf%20from%20that%20vertex.) it's the **height** of a graph. I am not sure that easy to understand to everybody.

So I propose:
"The rank of task represents the minimum number of levels or steps from the initial task of the CP (rank 0) to the current task"

This can be still simplified.

Reply:
The order here is in the mathematical definition of order, like here https://fr.wikipedia.org/wiki/Fonction_d%27ordre_sup%C3%A9rieur
I could add "the mathematical order" maybe ?

Reply:
I get that it's may not be perfectly true,  but IMO it's the idea to have in minde when dealing with ranks no ?

Reply:
And from the wiki it would the "depth" no ?

Reply:
yes my bad it's the depth not the height.